File: elixir/test/output_activate_digital_module_entity_test.exs
# OutputActivateDigitalModule entity test (offline, mock transport)

defmodule BluefinTecsUserBackoffice.OutputActivateDigitalModuleEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinTecsUserBackoffice.Helpers, as: H
  alias BluefinTecsUserBackoffice.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/output_activate_digital_module/OutputActivateDigitalModuleTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinTecsUserBackoffice.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.output_activate_digital_module"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinTecsUserBackoffice.test()
    ent = BluefinTecsUserBackoffice.output_activate_digital_module(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = BluefinTecsUserBackoffice.test(S.jm(["entity", S.jm(["output_activate_digital_module", S.jm([])])]))
    ent = BluefinTecsUserBackoffice.output_activate_digital_module(sdk)
    made = BluefinTecsUserBackoffice.Entity.OutputActivateDigitalModule.create(ent, S.jm(["name", "test-create"]))
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end

File: scala/sdktest/OutputActivateDigitalModuleEntityTest.scala
// Generated basic-flow test for the output_activate_digital_module entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped OutputActivateDigitalModuleTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsuserbackofficesdk.core.{Helpers, BluefinTecsUserBackofficeSDK}
import voxgig.bluefintecsuserbackofficesdk.utility.struct.Struct

object OutputActivateDigitalModuleEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("output_activate_digital_module.instance") {
      val testsdk = BluefinTecsUserBackofficeSDK.testSDK()
      val ent = testsdk.outputActivateDigitalModule(null)
      rep.check("output_activate_digital_module.instance", ent != null, "expected non-null output_activate_digital_module entity")
    }

    rep.scope("output_activate_digital_module.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/output_activate_digital_module/OutputActivateDigitalModuleTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsUserBackofficeSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("output_activate_digital_module01", "OUTPUT_ACTIVATE_DIGITAL_MODULE01")
      idmap.put("output_activate_digital_module02", "OUTPUT_ACTIVATE_DIGITAL_MODULE02")
      idmap.put("output_activate_digital_module03", "OUTPUT_ACTIVATE_DIGITAL_MODULE03")
      val now = System.currentTimeMillis()

      // CREATE
      val outputActivateDigitalModuleRef01Ent = client.outputActivateDigitalModule(null)
      var outputActivateDigitalModuleRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.output_activate_digital_module"), "output_activate_digital_module_ref01"))
      val outputActivateDigitalModuleRef01DataResult = outputActivateDigitalModuleRef01Ent.create(outputActivateDigitalModuleRef01Data, null)
      outputActivateDigitalModuleRef01Data = Helpers.toMapAny(outputActivateDigitalModuleRef01DataResult)
      rep.check("output_activate_digital_module.create.map", outputActivateDigitalModuleRef01Data != null, "expected create result to be a map")
    }
  }
}
